provider_shopper/test/cart_widget_test.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider/provider.dart';
+import 'package:provider_shopper/models/cart.dart';
+import 'package:provider_shopper/models/catalog.dart';
+import 'package:provider_shopper/screens/cart.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  Widget cartScreen() => MultiProvider(
+        providers: [
+          Provider(create: (context) => CatalogModel()),
+          ChangeNotifierProxyProvider<CatalogModel, CartModel>(
+            create: (context) => CartModel(),
+            update: (context, catalog, cart) {
+              cart.catalog = catalog;
+              return cart;
+            },
+          ),
+        ],
+        child: MaterialApp(
+          home: MyCart(),
+        ),
+      );
+  group('Cart Screen basic widget testing', () {
+    testWidgets('Testing the layout of the widgets', (tester) async {
```

Comment:
Testing the entire basic layout of the screen is likely unnecessary. Let's be honest -- it would be *super* weird if that behavior were to suddenly change. 😄 

What you could test, though, is the functionality that makes CartScreen *interesting*. You're doing that below when you check to see if the BUY button causes the SnackBar to appear. Most screens don't have a button with that label that cause a snackbar to appear, but CartScreen does, so you test *that*. 

Similarly, what you could do here is create two tests, one with an empty cart and one with a full one, and just make sure the text for the total cost and number of rows are correct. *That's* work that CartScreen is doing that's specific to it.